drive/modules/listener/watch_schedular.bal

```diff
@@ -45,19 +53,19 @@ class Job {
             checkpanic validateSpecificFolderExsistence(self.config.specificFolderOrFileId.toString(), 
             self.driveClient);
             self.specificFolderOrFileId = self.config.specificFolderOrFileId.toString();
-            self.watchResponse = checkpanic self.driveClient->watchFilesById(self.specificFolderOrFileId.toString(), 
+            self.watchResponse = checkpanic watchFilesById(self.config.clientConfiguration, self.specificFolderOrFileId.toString(), 
             self.config.callbackURL);
             self.isWatchOnSpecificResource = true;
         } else if (self.config.specificFolderOrFileId is string && self.isFolder == false) {
             checkpanic validateSpecificFolderExsistence(self.config.specificFolderOrFileId.toString(), 
             self.driveClient);
             self.specificFolderOrFileId = self.config.specificFolderOrFileId.toString();
-            self.watchResponse = checkpanic self.driveClient->watchFilesById(self.specificFolderOrFileId.toString(), 
+            self.watchResponse = checkpanic watchFilesById(self.config.clientConfiguration, self.specificFolderOrFileId.toString(), 
```

Comment:
When we use `checkpanic` it will throw a runtime error and task will get stop from there onwards. I guess if failed, we need to keep trying in a controlled way. 

```
run () {

    try {

     //logic

    }  catch (Throwable e) {
       
       sleep(5000)
       //schedule new job to run immediately
       
    }
    
    //schedule new job to run timely 

}
```

Reply:
Please address in a different PR.